plugins/identities/http.go

```diff
@@ -148,6 +148,20 @@ func (c *HTTPClient) GetUser(userID string) (interface{}, error) {
 	return res, nil
 }
 
+func (c *HTTPClient) IsActiveUser(userID string) (bool, error) {
+	userDetails, err := c.GetUser(userID)
+	if err != nil {
+		return false, err
+	}
+
+	if userDetailsMap, ok := userDetails.(map[string]interface{}); ok {
+		if accountStatus, ok := userDetailsMap[accountStatus]; ok {
```

Comment:
@singhvikash11 This `accountStatus ` key is hardcoded. We can not assume that IAM will always have this as key. They can actually have a different key to denote this. Also Guardian should decouple this from IAM I think. Guardian can manage its status for a user if needed and use that to revoke access. The status in Guardian can then be used to refresh from IAM and revoke access if needed. WDYT?

Reply:
slightly disagreement here and I am visualising this as part of the first implementation:
`iam:
  config:
   provider: http
  schema:
      account_status: status == 'InActive'`

Reply:
instead, we can add new config for determining account status field, e.g.:
```yaml
iam:
  provider: http
  config:
    url: ...
    schema:
      name: fullName
      foo: bar
      ...
    account_status_key: is_active # boolean expected
    # OR
    account_status: status == 'active'
```

with user object example as following
```json
{
  "fullName": "John Doe",
  "status": "active",
  "is_active": true,
  ...
}
```

Reply:
I like this approach, should I go ahead and implement this

@rahmatrhd @ravisuhag wdyt?

Reply:
@singhvikash11 can you update the changes discussed so far on the issue to cross-check the final approach?

Reply:
@rahmatrhd @ravisuhag As we discussed, the approach should be:

- A cron job to trigger this flow.
- Job will fetch all active appeals and group them based on the appeal's user, policy_id and policy_version.
- For each group, it'll evaluate user account status as either `true` or `false` based on the status expression configure in the policy.
- For each user with account status `false`, the job will revoke all appeals for that user. 

The job will show o/p like:

> INFO[0000] running inactive user appeal revoke job      
INFO[0001] successful appeal revocation                  count=0 ids="[]" user=vikash.singh@gojek.com
INFO[0001] failed appeal revocation                      count=0 ids="[]" user=vikash.singh@gojek.com

> INFO[0000] running inactive user appeal revoke job      
INFO[0003] appeal revoked                                id=e094bbec-e826-469f-9685-c780c5de76d8
INFO[0003] successful appeal revocation                  count=1 ids="[e094bbec-e826-469f-9685-c780c5de76d8]" user=vikash.singh@gojek.com
INFO[0003] failed appeal revocation                      count=0 ids="[]" user=vikash.singh@gojek.com

Reply:
Sample policy:

>
{
    "id": "sample_policy",
    "steps": [],
    "iam": {
        "provider": "http",
        "config": {
            },
        "schema": {
            "job_title": "title",
            "manager_email": "managerEmail"
        },
        **"account_status":"$user.accountStatus == 'active'"**
    }
}